src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/TagHelperResolutionResultJsonConverter.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Razor.Language;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Razor.Serialization
+{
+    internal class TagHelperResolutionResultJsonConverter : JsonConverter
+    {
+        public static readonly TagHelperResolutionResultJsonConverter Instance = new TagHelperResolutionResultJsonConverter();
+
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(TagHelperResolutionResult).IsAssignableFrom(objectType);
```

Comment:
Do a strict equals comparison. It's faster. We don't do that in TagHelper land only because there can be mock TagHelpers